### ISSUE TITLE
MachinePools/VMSS NetworkInterfaces wrong configuration

### ISSUE THREAD

**Issue by mweibel** [kubernetes-sigs/cluster-api-provider-azure#3224]
/kind bug

**What steps did you take and what happened:**
We have a bunch of MachinePools with separate subnets. The corresponding code for setting up the NetworkInterfaces changed substantially in #2411. 
The code in question breaks CNI due to not setting enableIPForwarding if we have a networkInterface specified but can't use the default due to a separate subnetName:
```
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: AzureMachinePool
metadata:
  name: REDACTED
  namespace: REDACTED
spec:
  template:
    networkInterfaces:
    - subnetName: REDACTED-subnet
```

Will result in the following networkInterface:
```
      "networkProfile": {"networkInterfaceConfigurations":[{"name":"REDACTED","properties":{"primary":true,"enableAcceleratedNetworking":false,"disableTcpStateTracking":false,"dnsSettings":{"dnsServers":[]},"enableIPForwarding":false,"ipConfigurations":[{"name":"private-ipConfig-0","properties":{"primary":true,"subnet":{"id":"/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.Network/virtualNetworks/REDACTED-vnet/subnets/REDACTED-subnet"},"privateIPAddressVersion":"IPv4"}}]}}]},
```

On a cluster built created with the old code, this is set to true. Also notice the `enableAcceleratedNetworking` is not set to true, even if the instance would support it and the docs state that it should be set to true if the VM type supports it. 

This probably breaks the CNI (we just built a fixed version of the controller to verify).

**What did you expect to happen:**
EnableIPForwarding set to true
AcceleratedNetworking set to true

**Anything else you would like to add:**
The [code in question](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/scalesets/scalesets.go#L561) duplicates a lot of code from [code right above](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/scalesets/scalesets.go#L511) and misses setting a few things. 
I guess the code would need to be refactored to better match what is done for other machines and unified as to not have two code parts working on the same logic.

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by CecileRobertMichon** [comment]
@mweibel I noticed the same inconsistencies while trying to fix #3163 and tried to refactor the NIC VMSS code in #3188, PTAL

That PR is currently blocked on a failing test... I could extract the refactor out of that PR and tackle the ipv6 part separately if that helps

**Comment by mweibel** [comment]
@CecileRobertMichon oh that's great news! I left a review. Let me know if anything's unclear :+1:
